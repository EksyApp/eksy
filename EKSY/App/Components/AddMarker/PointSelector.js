import React, {Component} from 'react'
import MapView from 'react-native-maps'

class PointSelector extends Component {
	
	constructor(props) {
		super(props)
		
		this._mapIsBeingMoved = false
		
		this.state = {
			markerLocation: {
				latitude: this.props.currentRegion.latitude,
				longitude: this.props.currentRegion.longitude
			}
		}
	}
	
	
	_handleChange(region) {
		this._mapIsBeingMoved = true
		this.setState({markerLocation: {latitude: region.latitude, longitude: region.longitude}})
	}
	
	_handleChangeComplete(region) {
		if (this._mapIsBeingMoved) {
			this.setState({markerLocation: {latitude: region.latitude, longitude: region.longitude}})
			this.props.onChange(region)
		}
		this._mapIsBeingMoved = false
	}
	
	render() {
		return (
				<MapView
						style={this.props.style}
						initialRegion={this.props.currentRegion}
						onRegionChange={(region) => this._handleChange(region)}
						onRegionChangeComplete={(region) => this._handleChangeComplete(region)}
				>
					<MapView.Marker coordinate={this.state.markerLocation}/>
				</MapView>
		)
	}
	
}

export default PointSelector
