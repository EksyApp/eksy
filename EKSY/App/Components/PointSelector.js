import React, {Component} from 'react'
import MapView from 'react-native-maps'
import PostOffice from '../lib/PostOffice'
import MapManager from '../Map/MapManager'

class PointSelector extends Component {

  constructor(props) {
    super(props)

    this.state = {
      markerLocation: {
        latitude: this.props.currentRegion.latitude,
        longitude: this.props.currentRegion.longitude
      }
    }
  }


  _handleChange(region){
    this.setState({markerLocation: {latitude: region.latitude, longitude: region.longitude}})
    this.props.onChange(region)
  }

  render() {
    return(
      <MapView
        style={this.props.style}
        initialRegion = {this.props.currentRegion}
        onRegionChange = {(region) => this._handleChange(region)}
      >
        <MapView.Marker coordinate={this.state.markerLocation}/>
      </MapView>
    )
  }
}

export default PointSelector
