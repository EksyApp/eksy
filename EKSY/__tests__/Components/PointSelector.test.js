import ReactNative, { StyleSheet } from 'react-native';
import React from 'react';
import PointSelector from '../../App/Components/Common/PointSelector';
import renderer from 'react-test-renderer';

jest.mock('react-native-fetch-blob', () => {
    return {
      DocumentDir: () => {},
      polyfill: () => {}
    }
  })

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children)
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children)
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children)
    }
  }

  class MockAnimatedRegion extends React.Component {
    render() {
      return React.createElement('AnimatedRegion', this.props, this.props.children)
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockAnimatedRegion.propTypes = MapView.AnimatedRegion.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  MockMapView.AnimatedRegion = MockAnimatedRegion;
  return MockMapView;
});

test('pointselector renders correctly', () => {
  const tree = renderer.create(
    <PointSelector initialRegion={{latitude: 1, longitude: 1, latitudeDelta: 1, longitudeDelta: 1}} style={StyleSheet.create({width: 10, height: 10})}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
