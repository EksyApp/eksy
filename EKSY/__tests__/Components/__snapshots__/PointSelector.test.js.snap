// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pointselector renders correctly 1`] = `
<MapView
  initialRegion={
    Object {
      "latitude": 1,
      "latitudeDelta": 1,
      "longitude": 1,
      "longitudeDelta": 1,
    }
  }
  onRegionChange={[Function]}
  onRegionChangeComplete={[Function]}
  style={
    Object {
      "height": 10,
      "width": 10,
    }
  }
>
  <Marker
    coordinate={
      Object {
        "latitude": 1,
        "longitude": 1,
      }
    }
  />
</MapView>
`;
